### ISSUE TITLE
Follow request profile link should go directly to the original page

### ISSUE THREAD

**Issue by fluffy-critter** [mastodon/mastodon#27234]
### Pitch

When looking at my follow requests, I want to be able to easily get to the requestor's original page, rather than the page that's run through my local instance's view first. The local view is usually going to be very limited in terms of which toots and behaviors are visible, and there doesn't seem to be any reason to default to showing that view for the purpose of vetting new followers.

### Motivation

I would like to reduce the number of clicks and new tabs/browsers/etc. it takes to vet a new follower.

**Comment by fluffy-critter** [comment]
> I wonder if we'd be better off adding a big "Review remote profile" button in the Approve/Decline modal on the page?

What I was hoping for would just be to have the link on the approval list go to the remote instance, and I guess that could open in a new tab by default? And then I do the approval locally, of course. I don't see any reason for added UX when the link could just, y'know, go to the place you need to look at.

**Comment by brendanjones** [comment]
Given the problem is a lack of profile content, wouldn't this be better solved by https://github.com/mastodon/mastodon/issues/34? (...if that ever gets solved, of course 😬 )

**Comment by fluffy-critter** [comment]
How would that interact with the remote instance being silenced, which is the main reason for follow requests happening in the first place? (In my experience anyway)

**Comment by brendanjones** [comment]
I don’t have the knowledge to answer that, I’m afraid, @fluffy-critter. Was simply pointing out that the proposed solution is potentially not the optimal solution. But I’m also well aware of the unlikelihood of the optimal solution being solved soon, so …

**Comment by ChaelCodes** [comment]
Even if we make it clear that this is an external link, the user can't follow from the external server, and there's no in-app navigation to get back to a place they can follow from.

We could have the link open in a new tab on the website, but that leaves the SPA and app experience unsolved.

I made the PR but left it in Draft because it's a very unsatisfying experience as-is.

**Comment by fluffy-critter** [comment]
Yes, I'm aware that the follow approval needs to be completed locally, which is why I'd suggested that the link be opened in a new tab. The current flow for doing a follow approval from another server requires opening multiple tabs and burrowing down into non-obvious menus, and by just opening up one additional tab and then going back to your follow request list to click the checkbox is way better than the current flow.

It's not *perfect* but it's better than what exists now.

**Comment by ke352802081770314** [comment]
in the /relationships list (where it is the other way round - it only links to the remote/original profile when some use cases need to open the user page on my local instance) I just made the suggestion in #26523 to provide both links. This  could help here too.

**Comment by guest20** [comment]
Linking to the original profile would be even more useful when the admins of your instance "restrict" an  instance or its users because follows from those users get dumped into "follow requests" even if you don't have your account locked, and then when you visit the "local" page for that user their profile is masked by a "your admins are mad at this user", so you have to dismiss that to see the profile and accept the follow request

1. See notification in side bar: 
<img width="74" alt="follow requests with a (1 unread) marker" src="https://github.com/user-attachments/assets/fd9a71bc-e08b-4308-90de-886a46634151" />

2. Visit the follow requests page:
<img width="136" alt="A single heavily redacted follow request" src="https://github.com/user-attachments/assets/e257a4dc-b1dc-411d-814e-7a23c6ea0a18" />

3. Find out who this is, by clicking their handle, and visit the local version of their page:
<img width="142" alt="the same heavily redacted profile, but showing a 'this account has been restricted by the admins of your instance' message in front of it" src="https://github.com/user-attachments/assets/2f19c6ad-1c54-4292-b705-fe2ca26b12ba" />

4. Click "show profile anyway" and actually look at the text etc ... 

These screens are from Mastodon: v4.4.3

**Comment by ThisIsMissEm** [comment]
This is functionality working as expected. You generally do not want to be taking users' off-server because of trust and safety considerations (e.g., phishing). You could offer another button from the follow request that is a "open original page" from that list. We have that UI elsewhere, and they people would know that they are leaving their server. 

Edit: I don't speak for Mastodon, but that's my opinion on this matter.

**Comment by guest20** [comment]
With this forced approval workflow, I must view the followers' profile in order to address the unread counter in the menu. I can't select "just let them follow me, my profile has already specified I do not care" - so I have to view the profile and make the decision if I want the bubble to go away. I could just click the green tick, but the moderators of my instance must have done this for some reason, so I have to check, right?

Having a second "this guy is on the sketchy instances list" warning shoved in my way when I am trying to address a follower request I only have because this "this guy is on the sketchy instances list" feels like it's creating "bureaucracy", not "trust" or "safety"

**Comment by fluffy-critter** [comment]
Absolutely agreed. If I could even just have *the option* to open the profile, even in another tab, without having to go through a half-dozen clicks, it'd be *way* less annoying. As it is I end up having to open up multiple browser tabs anyway and the flow is just obnoxious.
